src/averageAges.js

```diff
@@ -15,13 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+  const males = people.filter(person => century
+    ? (Math.ceil(person.died / 100) === century && person.sex === 'm')
+    : person.sex === 'm'
+  );
+  const maleAges = males.map(person => person.died - person.born);
+  const maleAgesSum = maleAges.reduce((sum, age) => sum + age, 0);
+  const averageMaleAges = maleAgesSum / maleAges.length;
```

Comment:
Also this logic was used in next future, so will be better to move it to separate function